birdhouse/components/weaver/post-docker-compose-up

```diff
@@ -333,6 +333,34 @@ for prov in ${WEAVER_WPS_PROVIDERS}; do
 done
 echo "${PREFIX}All Weaver remote WPS providers registered successfully!"
 
+if [ x"${WEAVER_UNREGISTER_DROPPED_PROVIDERS}" = x"True" ]; then
+    # Get all registered providers whether they are working or not
+    all_providers_resp=$( \
+        curl_cmd --insecure --silent --location \
+            -m ${REQUEST_TIMEOUT} \
+            -b "${cookie}" \
+            "${WEAVER_URL}/providers?check=false&detail=false" \
+    )
+    # Get all registered working providers
+    working_providers_resp=$( \
+        curl_cmd --insecure --silent --location \
+            -m ${REQUEST_TIMEOUT} \
+            -b "${cookie}" \
+            "${WEAVER_URL}/providers?check=true&detail=false" \
+    )
+    working_providers=$(echo "$working_providers_resp" | sed -e 's/.*"providers":\[//' -e 's/\].*//' -e 's/[",]/ /g')
+    for prov in $(echo "$all_providers_resp" | sed -e 's/.*"providers":\[//' -e 's/\].*//' -e 's/[",]/ /g'); do
+        if echo "${WEAVER_WPS_PROVIDERS}" | grep -wqv "$prov" && echo "${working_providers}" | grep -wqv "$prov"; then
```

Comment:
Might have an issue with `-` that is not considered in the "word" characters. 

Example: `finch` replaced by some kind of `finch-new finch-old` definition. If `finch` is disabled, and therefore not in the list of enabled & working providers, `echo "finch-1 finch-2" | grep -wv "finch"` (error/if not triggered) believes that `finch` is still there and won't remove the provider.

Reply:
Good catch. Thanks!